exporter/prometheusremotewriteexporter/factory.go

```diff
@@ -96,7 +90,7 @@ func createDefaultConfig() component.Config {
 		RemoteWriteQueue: RemoteWriteQueue{
 			Enabled:      true,
 			QueueSize:    10000,
-			NumConsumers: 5,
```

Comment:
I believe NumConsumers is used for concurrent requests here: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/83f1fc67bd14584583646512748726047997d02f/exporter/prometheusremotewriteexporter/exporter.go#L259

Seems like this would decrease parallelism in the default case. Is there a way for us to do this in a way that is less likely to be breaking?  Maybe set the export concurrency to `max(5, NumConsumers)`, so we don't reduce parallelism in that part of the exporter?

Reply:
Unless i was reading it wrong, you maintain the same parallelism anyway.
Since while the default was at 5, it ALWAYS overridden in L64 to 1.
Now I am just setting the default to 1. And the concurrency will remain the same min(1, len(requests)).
You never had concurrency it in the first place - what you had was a simple chunk logic, to split a large request into smaller requests and send them sequentially.

Now you have proper parallelism since its no longer hard-coded to 1.

Previous:
Consumers 5 (hard-coded at 1), len(requests) at 2 = Parallelism 1


Now:
Consumers 1 (default), len(requests) at 2 = Parallelism 1
Consumers 5 (default), len(requests) at 2 = Parallelism 2
Consumers 5 (default), len(requests) at 10 = Parallelism 5

Reply:
The value is passed to newPRWExporter on line 45 above: `prwe, err := newPRWExporter(prwCfg, set)`.  This happens before we override NumConsumers on line 64.

My concerns are:

* Users are currently setting NumConsumers with the expectation that it will increase the number of chunks used.  E.g. users might set it to 20 to split requests more.  When we make this change, it will result in them sending with actual parallelism.
* Users can currently configure how requests are split, while still preventing out of order requests.  With this new design, they have to configure both together.

WDYT about having a separate configuration option for the request parallelism vs the number of consumers?

Reply:
Got it. Let me double check.

I am also ok with creating an additional configuration field (and setting at 5 to keep it default as is), but I am concerned with eventual "semantic confusion" of the fields (keeping num_consumers at 5, while creating another field to control parallelism), unless we make a breaking change in the configuration.

To keep compatibility, but it kinda goes against the `exporterhelper\queue` standard that "kinda" use here.
```
another_field_worker_count: X # this controllers "workers" count.
remote_write_queue:
  num_consumers: 5 # this controls individual request chunk split
```

Ideally, maybe something like this, but a breaking change in config.
```
max_batch_size_requests: 5 # we define this field to control chunk split, since its split by the `max_batch_size_bytes`
remote_write_queue:
  num_consumers: X  # also starts getting aligned with OTLP exporterhelper
```

wdyt?

Reply:
The second was what I had in mind.  We would need to feature-gate the change, but I think it is the right eventual direction.

Reply:
awesome.
im very okay with a feature gate to move towards standardization.

Ill push the change here in the next days. 

Cheers.

Reply:
hi @dashpole @ArthurSens 

reset changes to be behind featuregate, and updated docs. Please take a look